manifests/plugin/column_view.pp

```diff
@@ -0,0 +1,26 @@
+# = Foreman Column View
+#
+# This class installs the column view plugin and optionally manages the configuration file
+#
+# === Parameters:
+#
+# $columns:: an hash of columns to add to the configuration
+#
+class foreman::plugin::column_view (
+  Optional[Hash[String,Hash[Enum['title','after','content','title','conditional','eval_content','view','width','custom_method'],String,3,9]]] $columns = undef
+){
+  $config_file = "${foreman::plugin_config_dir}/foreman_column_view.yaml"
+
+  foreman::plugin { 'column_view':
+  }
+
+  if $columns {
+    file { $config_file:
```

Comment:
You can also use the ```$config``` parameter of ```foreman::plugin``` to achieve the same to manage this file. It'll even match the filename in by default so no need to override ```$config_file```.